blocksense-network/blocksense
# Implement allow list for outbound http requests to URLs

## Description

Implement allow list for outbound http requests.


## Changes

* Update wasmtime to v22 to match spin wasmtime version.
* Implement the trait `OutboundWasiHttpHandler` for our `HttpRuntimeData`

## How to test

Change these properties in `spin.toml` from revolut example folder in the sdk:
* `allowed_outbound_hosts` - try allowing another URL not revolut.
* `sequencer` - you can add postman echo for returning the same request. (https://postman-echo.com/get)
*  Make sure you've ran `.libs/sdk/build-and-install-plugin.sh` (it installs our oracle trigger)
- [x] `./libs/sdk/start-oracle-example.sh revolut-example`

## Checklist

- [x] The new changes/fixes are tested
